crates/eval/src/eval.rs

```diff
@@ -75,6 +76,17 @@ fn main() {
     app.run(move |cx| {
         let app_state = init(cx);
 
+        let system_id = ids::get_or_create_id(&ids::eval_system_id_path())
```

Comment:
You can use `.ok()` to convert a `Result` type into an `Option`, so you can avoid having to provide a default for the `unwrap_or_else`, following by converting it to an `Option`.

```rs
        let system_id = ids::get_or_create_id(&ids::eval_system_id_path()).ok();
        let installation_id = ids::get_or_create_id(&ids::eval_installation_id_path()).ok();
        let session_id = uuid::Uuid::new_v4().to_string();

        app_state
            .client
            .telemetry()
            .start(system_id, installation_id, session_id, cx);
```

Reply:
I suppose this will be different behavior, as yours will always be some string, and mine will result in there being potential Nones. Maybe you were always wanting strings, ignore if so!

Reply:
I like the idea of _maybe_ a none. I think that is smart!